src/gui/accountstate.cpp

```diff
@@ -468,6 +520,35 @@ void AccountState::slotOcsError(int statusCode, const QString &message)
     qCDebug(lcAccountState) << "Error " << statusCode << " while fetching new navigation apps: " << message;
 }
 
+void AccountState::slotCheckConnection()
+{
+    if (_lastCheckConnectionTimer.isValid()) {
+        const auto currentDelay = (retryCount() <= 1 ? ConnectionValidator::DefaultCallingIntervalMsec :
+                                                       (retryCount() == 2 ? ConnectionValidator::DefaultCallingIntervalMsec * 2 :
+                                                                            (retryCount() == 3 ? ConnectionValidator::DefaultCallingIntervalMsec * 4 :
+                                                                                                 ConnectionValidator::DefaultCallingIntervalMsec * 8)));
```

Comment:
This seems a little convoluted, no? Can we not just multiply by the retry count and limit the minimum and maximum multiplier with a qMin and a qMax?

Reply:
E.g.: 

```
const auto interval = ConnectionValidator::DefaultCallingIntervalMsec;
qMin( qMax( (retryCount() + retryCount()) * interval, interval ), interval * 8 );
```

Reply:
@mgallien @claucambra Agree, it is hard to read such construction. Here is my vision of how this can be done:
```
enum { DefaultCallingIntervalMsec = 62 * 1000, DefaultCallingIntervalMaxMsec = DefaultCallingIntervalMsec * 8 };

const auto interval = retryCount() <= 3 ? (retryCount() + 1) * ConnectionValidator::DefaultCallingIntervalMsec : DefaultCallingIntervalMaxMsec ;
```

Reply:
hopefully the new logic is better